data/sqlutil/macros.go

```diff
@@ -124,110 +124,73 @@ var DefaultMacros = Macros{
 	"column":     macroColumn,
 }
 
-type Macro struct {
-	Name string
-	Args []string
+type macroMatch struct {
+	full string
+	args []string
 }
 
 // getMacroMatches extracts macro strings with their respective arguments from the sql input given
 // It manually parses the string to find the closing parenthesis of the macro (because regex has no memory)
-func getMacroMatches(input string, name string) ([]Macro, error) {
-	macroName := fmt.Sprintf("\\$__%s\\b", name)
-	matchedMacros := []Macro{}
-	rgx, err := regexp.Compile(macroName)
+func getMacroMatches(input string, name string) ([]macroMatch, error) {
+	rgx, err := regexp.Compile(fmt.Sprintf(`\$__%s\b`, name))
 
 	if err != nil {
 		return nil, err
 	}
 
-	// get all matching macro instances
-	matched := rgx.FindAllStringIndex(input, -1)
+	var matches []macroMatch
+	for _, window := range rgx.FindAllStringIndex(input, -1) {
+		start, end := window[0], window[1]
+		args, length := parseArgs(input[end:])
+		if length < 0 {
+			return nil, fmt.Errorf("failed to parse macro arguments (missing close bracket?)")
+		}
+		matches = append(matches, macroMatch{full: input[start : end+length], args: args})
+	}
+	return matches, nil
+}
 
-	if matched == nil {
-		return nil, nil
+// parseArgs looks for a bracketed argument list at the beginning of argString.
+// If one is present, returns a list of whitespace-trimmed arguments and the
+// length of the string comprising the bracketed argument list.
+func parseArgs(argString string) ([]string, int) {
+	if !strings.HasPrefix(argString, "(") {
+		return nil, 0 // single empty arg for backwards compatibility
 	}
 
-	for matchedIndex := 0; matchedIndex < len(matched); matchedIndex++ {
```

Comment:
The logic from here to the end of the function determines what part of the `input` string represents arguments to the current macro. `parseArgs` below implements most of the same logic in a slightly different way but also splits the argument string. I've removed the redundant functionality in `getMacroMatches` and updated `parseArgs` to do all the necessary work of parsing arguments.